src/esp/gfx/replay/Keyframe.h

```diff
@@ -55,6 +55,7 @@ struct Keyframe {
   std::vector<std::pair<RenderAssetInstanceKey, RenderAssetInstanceState>>
       stateUpdates;
   std::unordered_map<std::string, Transform> userTransforms;
+  Cr::Containers::Optional<std::vector<LightInfo>> lights;
```

Comment:
I'm a little torn on serializing our entire quirky `LightInfo` struct (including LightPositionModel). I talked in the design doc about wanting gfx-replay to be agnostic to the renderer, and most renderers IMO won't offer "object-local" or "camera-local" lights. I think we only need to support lights specified in global space. Also, I don't think anybody is using object-local or camera-local lights in Habitat. I don't think we'll bother implementing them in the new batch renderer.

See also this discussion here: https://cvmlp.slack.com/archives/G0131KVLBLL/p1660649432149979 .

It is probably fine to leave this as-is, but let's beware introducing unneeded complexity into the gfx-replay format in general.

Reply:
I completely agree.

I feel like it doesn't hurt to leave the `LightPositionModel` field as-is for now. We should look into specifying a standard light data layout as we integrate them with the batch renderer.

I am still unfamiliar with the batch renderer, but if we are following a pure ECS approach, we might want to separate the transform component from the light data.

>I think we only need to support lights specified in global space.

Object-local lights may be occasionally useful (e.g. if the simulated agent has a light). However, I don't think that we are doing anything like that at the moment.
However, I don't know how camera-local lights would be useful.